Document `gp validate` (#3290)
gitpod-io/website: gitpod/docs/configure/workspaces/index.md

# Workspaces

> Gitpod is not simply "moving your laptop into the cloud". A key benefit of using a Cloud Development Environment ([CDE](/cde)) is _reproducibility_. When your workspace is configured, opening a new workspace is effortless—allowing you to fully embrace ephemeral development environments.

Reviewer: Really like this intro 🔥 If I'm being pedantic, should we claim that it's "the key benefit"? Or should we say "One of the big benefits..."

Reviewer 2: ```suggestion
Gitpod is not simply "moving your laptop into the cloud". One key benefit of using a Cloud Development Environment ([CDE](/cde)) is _reproducibility_. When your workspace is configured, opening a new workspace is effortless—allowing you to fully embrace ephemeral development environments.
```

ports:
  - port: 3000
    onOpen: open-preview
    name: Website
    description: Website Preview
```

**Caption:** An example project configured to install, build and run a `yarn` project with a webserver, exposed on port 3000. On start, the webserver preview is opened automatically.

### The workspace image

In addition to the `gitpod.yml` you can also specify a workspace image for:

1. Improved caching and performance
2. Application portability
3. Re-using an existing Dockerfile

Currently, Gitpod only supports Docker for workspace images. The Dockerfile can either be kept alongside your Gitpod configuration, or you can consume an existing public, or private image.

See [Workspace Image](/docs/configure/workspaces/workspace-image) for more.

## Creating a Gitpod configuration

You can create a `.gitpod.yml` manually, or by using the `gp init` command (or `gp init -i` for interactive mode). The `gp` CLI tool is part of the [Gitpod CLI](/docs/references/gitpod-cli), which is included in all Gitpod workspaces by default.

```sh
gp init
```

See the [Gitpod CLI](/docs/references/gitpod-cli) page for more.

## Validate your Gitpod configuration

> You can test your configuration including your `.gitpod.yml` without leaving your worksapce or committing you changes by using the `gp rebuild` command. You can think of what this command does as opening a workspace that runs from within your current workspace, allowing you to troubleshoot workspace configuration (ports, tasks, etc) and more.

Reviewer 3: I love the *thus* in @laushinka's suggestion (English is so 🤌 sometimes 😆). I love the suggestion, just would replace the second that with a *which* (to not have it twice in the same sentence):
```suggestion
You can test your configuration, including your `.gitpod.yml`, without leaving your workspace or committing your changes by using the `gp rebuild` command. This command opens a workspace (that runs from within your current workspace) which includes your configuration changes. Thus, allowing you to troubleshoot workspace configuration (ports, tasks, etc.) and more.
```

> You can test your configuration including your `.gitpod.yml` without leaving your worksapce or committing you changes by using the `gp rebuild` command. You can think of what this command does as opening a workspace that runs from within your current workspace, allowing you to troubleshoot workspace configuration (ports, tasks, etc) and more.

Reviewer 4: That looks much better 👍 . I can only suggest a change for a sentence:

From:
> This command opens a workspace (that runs from within your current workspace) that includes your configuration changes

To:

> This command opens a workspace that includes your configuration changes from within your current workspace.

And maybe we can also mention something like below:?

> This is similar to running docker-in-docker.

- [Workspace Image](/docs/configure/workspaces/workspace-image)
- [Tasks](/docs/configure/workspaces/tasks)
- [Ports](/docs/configure/workspaces/ports)

> - [Testing Workspace Configuration Changes](/docs/configure/workspaces/testing-changes)

Reviewer 2: We should [update the gitpod configuration page](https://www.gitpod.io/docs/introduction/learn-gitpod/gitpod-yaml) to talk about the inner loop, and how to update and test configuration changes, referencing this documentation.
